dvc/repo/plots/__init__.py

```diff
@@ -260,6 +302,168 @@ def _collect_plots(
     return result
 
 
+def _get_data_targets(pathlib, definitions: Dict):
+    result: Set = set()
+    if "data" in definitions:
+        for file_relpath, content in definitions["data"].items():
+            if "data" in content:
+                for plot_id, config in content["data"].items():
+                    result = result.union(
+                        infer_data_sources(
+                            pathlib, file_relpath, plot_id, config
+                        )
+                    )
+    return result
+
+
+def infer_data_sources(pathlib, config_path, plot_id, config):
+    import posixpath
+
+    def _deduplicate(lst: List):
+        return list({elem: None for elem in lst}.keys())
+
+    y = config.get("y", None)
+    if isinstance(y, dict):
+        sources = list(y.keys())
+    else:
+        sources = [plot_id]
+
+    # TODO bug in dvc_objects? normpath normalizes to windows path on windows
+    return _deduplicate(
+        posixpath.normpath(pathlib.join(pathlib.dirname(config_path), source))
+        for source in sources
+    )
+
+
+def _matches(targets, config_file, plot_id):
+    import re
+
+    from dvc.utils.plots import get_plot_id
+
+    if not targets:
+        return True
+
+    full_id = get_plot_id(plot_id, config_file)
+    if any(
+        (re.match(target, plot_id) or re.match(target, full_id))
+        for target in targets
+    ):
+        return True
+    return False
+
+
+def _dvcfile_relpath(repo, dvcfile):
+    fs = repo.dvcfs
+
+    # TODO from_os_path changes abs to relative
+    return fs.path.relpath(
+        fs.path.join("/", fs.from_os_path(dvcfile.path)), fs.path.getcwd()
+    )
+
+
+def _collect_output_plots(
+    repo, targets, props, onerror: Optional[Callable] = None
+):
+    fs = repo.dvcfs
+    result: Dict[str, Dict] = {}
+    for plot in repo.index.plots:
```

Comment:
Not blocker of this P.R.

I think flexible plots should be included as part of the `repo.index` somehow

Reply:
https://github.com/iterative/dvc/issues/7932